src/Essentials/src/SecureStorage/SecureStorage.android.cs

```diff
@@ -18,22 +19,28 @@ Task<string> PlatformGetAsync(string key)
 				{
 					lock (locker)
 					{
-						return GetEncryptedSharedPreferences().GetString(key, null);
+						ISharedPreferences sharedPreferences = GetEncryptedSharedPreferences();
+						if (sharedPreferences != null)
+							return sharedPreferences.GetString(key, null);
+
+						System.Diagnostics.Debug.WriteLine(
+							$"Unable to decrypt key, {key}, which is likely due to key corruption. Removing old key and returning null.");
+						return null;
 					}
 				}
-				catch (AEADBadTagException)
+				catch (GeneralSecurityException)
 				{
 					// TODO: Use Logger here?
-					System.Diagnostics.Debug.WriteLine($"Unable to decrypt key, {key}, which is likely due to an app uninstall. Removing old key and returning null.");
-					Remove(key);
```

Comment:
What happened to the removes?

Reply:
Yeah, I think I missed that one 🤔 Will add it back where appropriate. Thanks!